pkg/provisioner/ironic/ironic.go

```diff
@@ -1165,10 +1165,10 @@ func (p *ironicProvisioner) GetFirmwareComponents() ([]metal3api.FirmwareCompone
 	// Get the components from Ironic via Gophercloud
 	componentList, componentListErr := nodes.ListFirmware(p.ctx, p.client, ironicNode.UUID).Extract()
 
-	if componentListErr != nil {
+	if componentListErr != nil || len(componentList) == 0 {
 		bmcAccess, _ := p.bmcAccess()
-		if bmcAccess.FirmwareInterface() == "no-firmware" {
-			return nil, fmt.Errorf("node %s is using firmware interface %s: %w", ironicNode.UUID, bmcAccess.FirmwareInterface(), componentListErr)
+		if bmcAccess.FirmwareInterface() != "redfish" {
+			return nil, fmt.Errorf("Driver does not support firmware updates")
```

Comment:
maybe log the unsupported driver name and ironic node uuid to give some more info as it was before

Reply:
maybe we can add the driver name, regarding the node uuid see Dmitry's comment https://github.com/metal3-io/baremetal-operator/pull/1559#discussion_r1553505273

Reply:
ok, I see his point
adding the driver name I think would still be useful

Reply:
ack, I will update this part o/